src/celeritas/optical/surface/SurfaceTraversalView.hh

```diff
@@ -0,0 +1,235 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/surface/SurfaceTraversalView.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/math/ArrayUtils.hh"
+#include "celeritas/optical/Types.hh"
+
+#include "SurfacePhysicsData.hh"
+#include "SurfacePhysicsUtils.hh"
+
+namespace celeritas
+{
+namespace optical
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Manage one-dimensional logic for traversing an optical surface.
```

Comment:
I'd add a few bullets about how the traversal proceeds over the surface crossing:
- *Always* starts at position 0 in the forward direction and does not relate to the surface data layout
- The position corresponds to the surface layers

An ASCII art drawing of the pre/post volume, traversal direction, etc. would be extremely helpful.

Might also be worth mentioning that the only params data needed are the number of subsurfaces in the surface being crossed.